mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp

```diff
@@ -348,6 +353,9 @@ convertVectorType(const spirv::TargetEnv &targetEnv,
     }
 
     Type elementType = convertSubByteIntegerType(options, intType);
+    if (!elementType)
+      return nullptr;
```

Comment:
The bug is that previously, it was assumed that if we entered this code path, the integer type was unsupported because of being sub-byte, but it might be unsupported for some other reason?

Reply:
The bug was essentially that it assumed that `convertSubByteIntegerType` would always succeed, but this is not the case even prior to my change related to subbyte types.

Reply:
Thanks. If you want to, you could put this in the pull-request description so it ends up in the commit message, but that may be unnecessary.